experimental/Schemes/Tjurina.jl

```diff
@@ -0,0 +1,770 @@
+export tjurina_algebra
+export tjurina_number
+
+#export order
+export order_as_series
+
+
+export is_finitelydetermined
+export determinacy_bound
+export sharper_determinacy_bound
+
+export is_contact_equivalent
+
+## literature:
+##        Greuel, Lossen, Shustin: 'Introduction to Singularities and Deformations'
+
+##    for positive characteristic the following papers
+##        Greuel, Pham: 'Mather-Yau Theorem in Positive Characteristic'
+##        Boubakri, Greuel, Markwig: 'Invariants of Hypersurface Singularities in Positive Characteristic'
+
+
+################################################################################
+
+#####                           Tjurina algebra                            #####
+
+################################################################################
+
+@doc raw"""
+    tjurina_algebra(f::MPolyRingElem)
+
+Return the global Tjurina algebra of the affine hypersurface `V(f)`.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> f = x^3 - y^2;
+
+julia> tjurina_algebra(f)
+Quotient
+  of multivariate polynomial ring in 2 variables x, y
+    over rational field
+  by ideal (x^3 - y^2, 3*x^2, -2*y)
+```
+"""
+function tjurina_algebra(f::MPolyRingElem)
+  R = parent(f)
+  return MPolyQuoRing(R, ideal(R, f) + jacobian_ideal(f))
+end
+
+
+
+@doc raw"""
+    tjurina_algebra(X::AffineScheme{<:Field,<:MPolyQuoRing})
+
+Return the global Tjurina algebra of the affine scheme `X`, if `X` is a hypersurface.
+Return an error otherwise.
```

Comment:
Does it really "return" an error (and what does that mean)? Or does it raise an error / throw an exception?